target/hwaccess/hwAccessIntf.C

```diff
@@ -0,0 +1,84 @@
+#include <hwAccessIntf.H>
+#include "hwBaseAccess.H"
+#include "hwDirectAccess.H"
+#include "hwSbeAccess.H"
+
+namespace hwaccess
+{
+int HwAccessIntf::getCfamRegisters(TargetPtr_t target, uint32_t addr, uint32_t& val)
+{
+    if(target == nullptr)
+        return -1;
+
+    auto accessPtr = target->getAttr<TARGETING::ATTR_HW_ACCESS_PTR>();
+    HwBaseAccess* hwAccessPtr = reinterpret_cast<HwBaseAccess*>(static_cast<uintptr_t>(accessPtr));
+
+    if (hwAccessPtr != nullptr)
+    {
+        return hwAccessPtr->getCfam(target, addr, val);
+    }
+
+    return -1;
```

Comment:
In the HwBaseAccess and the other access classes, we return as soon the target pointer is nullptr, here if the hwAccessPtr is nullptr, it comes to the end and returns. Can we have a similar approach everywhere